src/utils/device/dehydration.ts

```diff
@@ -18,40 +18,37 @@ import { logger } from "matrix-js-sdk/src/logger";
 
 import { MatrixClientPeg } from "../../MatrixClientPeg";
 
-// the interval between creating dehydrated devices
-const DEHYDRATION_INTERVAL = 7 * 24 * 60 * 60 * 1000;
-
-// check if device dehydration is enabled
-export async function deviceDehydrationEnabled(): Promise<boolean> {
+/**
+ * Check if device dehydration is enabled.
+ *
+ * Dehydration can only be enabled if encryption is available, and the crypto
+ * backend supports dehydration.
+ *
+ * Dehydration can currently only enabled by setting a flag in the .well-known file.
+ */
+async function deviceDehydrationEnabled(): Promise<boolean> {
     const crypto = MatrixClientPeg.safeGet().getCrypto();
     if (!crypto) {
         return false;
     }
     if (!(await crypto.isDehydrationSupported())) {
         return false;
     }
-    if (await crypto.isDehydrationKeyStored()) {
-        return true;
-    }
     const wellknown = await MatrixClientPeg.safeGet().waitForClientWellKnown();
     return !!wellknown?.["org.matrix.msc3814"];
 }
 
-// if dehydration is enabled, rehydrate a device (if available) and create
-// a new dehydrated device
-export async function initializeDehydration(reset?: boolean): Promise<void> {
+/**
+ * If dehydration is enabled, rehydrate a device (if available) and create
+ * a new dehydrated device.
+ *
+ * @param createNewKey: force a new dehydration key to be created, even if one
+ *   already exists.  This is used when we reset secret storage.
+ */
+export async function initializeDehydration(createNewKey: boolean = false): Promise<void> {
```

Comment:
not that we're very good at being consistent, but I think there are slightly more `initialise`s than `initialize`s in the codebase.

(And https://github.com/matrix-org/matrix-spec/blob/main/meta/documentation_style.rst asks for British English. Not that this is the spec.)